src/facebook_example/bot.clj

```diff
@@ -51,18 +63,30 @@
     (reaction/thank-for-attachment)))
 
 ; You should not need to touch the following code :)
-(defn postback? [messaging-event](contains? messaging-event :postback))
-(defn attachments? [messaging-event] (contains? (:message messaging-event) :attachments))
-(defn message? [messaging-event] (contains? messaging-event :message))
+(defn process-event [event]
+  ; The order of the matchers is important!
+  (match [event]
+    ; The user `sender-id` has selected one quick-reply option
+    [{:message {:quick_reply _}}]
+    (on-quick-reply event)
+
+    ; The user `sender-id` has sent a text message
+    [{:message {:text _}}]
+    (on-message event)
+
+    ; The user `sender-id` has sent a file or sticker
+    [{:message {:attachments _}}]
+    (on-attachments event)
```

Comment:
Oh sure, I did not know that.
Btw yes, they are matched in the same way routes usually are, so the order is important and supposed to be so.